src/main/java/com/example/Restaurant_Management_System_REST_API/service/ReportService.java

```diff
@@ -0,0 +1,392 @@
+
+        StreamingResponseBody stream = outputStream -> {
+
+            Workbook workbook = new XSSFWorkbook();
+            Sheet sheet = workbook.createSheet("restaurantOrdersByTimeRange");
+
+            // Create header row
+            Row headerRow = sheet.createRow(0);
+            stringSetter.setCellValue(headerRow, 0, "Restaurant Order ID");
+            stringSetter.setCellValue(headerRow, 1, "Order time");
+            stringSetter.setCellValue(headerRow, 2, "Order status");
+            stringSetter.setCellValue(headerRow, 3, "Table id");
+            stringSetter.setCellValue(headerRow, 4, "Telephone number");
+            stringSetter.setCellValue(headerRow, 5, "Total amount to pay");
+
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
+
+            // Fill data rows
+            for (int i = 0; i < restaurantOrderByOrderTimeRange.size(); i++) {
+                RestaurantOrder restaurantOrder = restaurantOrderByOrderTimeRange.get(i);
+                OrderStatus orderStatus = restaurantOrder.getOrderStatus();
+
+                Row row = sheet.createRow(i + 1);
+                longSetter.setCellValue(row, 0, restaurantOrder.getId());
+                stringSetter.setCellValue(row, 1, restaurantOrder.getOrderTime().format(formatter));
+                stringSetter.setCellValue(row, 2, orderStatus.toString());
+                longSetter.setCellValue(row, 3, restaurantOrder.getTable().getId());
+                stringSetter.setCellValue(row, 4, restaurantOrder.getTelephoneNumber());
+                doubleSetter.setCellValue(row, 5, restaurantOrder.getTotalAmountToPay());
+
+            }
+            workbook.write(outputStream);
+            workbook.close();
+        };
+
+        return stream;
+    }
+
+    public StreamingResponseBody getRestaurantOrderByOrderStatus(OrderStatus orderStatus) {
+
+        List<RestaurantOrder> restaurantOrderByOrderStatus = restaurantOrderRepository.findByOrderStatus(orderStatus);
+
+        StreamingResponseBody stream = outputStream -> {
+
+            Workbook workbook = new XSSFWorkbook();
+            Sheet sheet = workbook.createSheet("restaurantOrdersByOrderStatus");
+
+            Row headerRow = sheet.createRow(0);
+            stringSetter.setCellValue(headerRow, 0, "Restaurant Order ID");
+            stringSetter.setCellValue(headerRow, 1, "Order time");
+            stringSetter.setCellValue(headerRow, 2, "Order status");
+            stringSetter.setCellValue(headerRow, 3, "Table id");
+            stringSetter.setCellValue(headerRow, 4, "Telephone number");
+            stringSetter.setCellValue(headerRow, 5, "Total amount to pay");
+            stringSetter.setCellValue(headerRow, 6, "Menu record name");
+            stringSetter.setCellValue(headerRow, 7, "Portions amount");
+
```

Comment:
Here are the strings that are being used earlier:

```java
stringSetter.setCellValue(headerRow, 0, "Restaurant Order ID");
stringSetter.setCellValue(headerRow, 1, "Order time");
stringSetter.setCellValue(headerRow, 2, "Order status");
stringSetter.setCellValue(headerRow, 3, "Table id");
stringSetter.setCellValue(headerRow, 4, "Telephone number");
stringSetter.setCellValue(headerRow, 5, "Total amount to pay");
```

That's called magic numbers or magic strings in this particular case. If the decision come later to change header names it's a bad idea to search for those Strings in the code. It's better to use constants.

Reply:
but everywhere in all methods? therefore I need to write each String twice, as private static final instance variable like this: private static final String RESTAURANT_ID = "ID";
and then each as parameter, for all tables that would be a lot. is that really what I should suppose to do?

Reply:
okay I did this!